ext/opcache/ZendAccelerator.c

```diff
@@ -511,6 +510,15 @@ zend_string* ZEND_FASTCALL accel_new_interned_string(zend_string *str)
 	memcpy(ZSTR_VAL(s), ZSTR_VAL(str), ZSTR_LEN(s) + 1);
 	ZCSG(interned_strings).top = STRTAB_NEXT(s);
 
+finish:
+	/* Transfer CE_CACHE map ptr slot to new interned string.
+	 * Should only happen for permanent interned strings with permanent map_ptr slot. */
+	if (ZSTR_HAS_CE_CACHE(str)) {
+		ZEND_ASSERT(GC_FLAGS(str) & IS_STR_PERMANENT);
+		GC_SET_REFCOUNT(s, GC_REFCOUNT(str));
+		GC_ADD_FLAGS(s, IS_STR_CLASS_NAME_MAP_PTR);
+	}
+
```

Comment:
This is only necessary for calls from accel_copy_permanent_strings(). Right?

Reply:
I think that's the only place it can happen, though it shouldn't hurt to check for it in other cases either.

This is missing a `&& !ZSTR_HAS_CE(s)` check though -- no need to do this again if cache is already present.